subnet/deployment-generator/gen.js

```diff
@@ -0,0 +1,487 @@
+    volume='./xdcchain'+i.toString()+':/work/xdcchain'
+    config='${SUBNET_CONFIG_PATH}/subnet'+i.toString()+'.env'
+    compose_profile='machine'+machine_id.toString()
+    subnet_nodes[node_name] = {
+      'image': 'xinfinorg/xdcsubnets:latest',
+      'volumes': [volume, '${SUBNET_CONFIG_PATH}/genesis.json:/work/genesis.json'],
+      'restart': 'always',
+      'network_mode': 'host',
+      // 'env_file': ['${SUBNET_CONFIG_PATH}/common.env', config],
+      'env_file': [config],
+      'profiles': [compose_profile]
+    }
+
+  }
+  return subnet_nodes
+}
+
+function genBootNode(machine_id){
+  config='${SUBNET_CONFIG_PATH}/common.env'
+  machine='machine'+machine_id.toString()
+  bootnode = {
+    'image': 'xinfinorg/xdcsubnets:latest',
+    'restart': 'always',
+    'env_file': config,
+    'volumes': ['./bootnodes:/work/bootnodes'],
+    'entrypoint': ['bash','/work/start-bootnode.sh'],
+    'command': ['-verbosity', '6', '-nodekey', 'bootnode.key'],
+    'ports': ['20301:20301/tcp','20301:20301/udp'],
+    'profiles': [machine]
+  }
+  return bootnode
+}
+
+function genMainnet(machine_id){
+  config='${SUBNET_CONFIG_PATH}/common.env'
+  machine='machine'+machine_id.toString()
+  mainnet = {    
+    'image': 'xinfinorg/devnet:latest',
+    'restart': 'always',
+    'env_file': config,
+    'ports': ['40313:30303', '9555:8545', '9565:8555'],
+    'profiles': [machine]
+  }
+  return mainnet
+}
+
+function genServices(machine_id) {
+  config='${SUBNET_CONFIG_PATH}/common.env'
+  // machine='services_machine'+machine_id.toString()
+  machine='services'
+  frontend = {
+    'image': 'xinfinorg/subnet-frontend:latest',    
+    'restart': 'always',
+    'volumes': [`${config}:/app/.env.local`],
+    'ports': ['5000:5000'],
+    'profiles': [machine]
+  }
+  relayer = {
+    'image': 'xinfinorg/xdc-relayer:latest',
+    'restart': 'always',
+    'env_file': config,
+    'profiles': [machine]
+  }
+  stats = {
+    'image': 'xinfinorg/subnet-stats-service:latest',
+    'restart': 'always',
+    'env_file': config,
+    'volumes': ['./stats-service/logs:/app/logs'],
+    'ports': ['3000:3000'],
+    'profiles': [machine]
+  },
+  puppeth = {
+    'image': 'xinfinorg/xdcsubnets:latest',
+    'entrypoint': ['puppeth'],
+    'volumes': ['./puppeth:/root/.puppeth'],
+    'profiles': ['none']
+  }, 
+  bootnode=genBootNode(machine_id),
+  mainnet=genMainnet(machine_id),
+
+
+  services = {
+    'bootnode': bootnode,
+    'mainnet': mainnet,
+    'relayer': relayer,
+    'stats': stats,
+    'puppeth': puppeth,
+    'frontend': frontend,
+  }
+
+  return services
+}
+
+function genSubnetConfig(subnet_id, key, ip_1, network_id, secret){
+  key_name = `key${subnet_id}`
+  private_key = key[key_name]['PrivateKey']
+  port = 20303+subnet_id-1
+  rpcport = 8545+subnet_id-1
+  wsport= 8555+subnet_id-1
+  config = `
+INSTANCE_NAME=subnet${subnet_id}
+PRIVATE_KEY=${private_key}
+BOOTNODES=enode://cc566d1033f21c7eb0eb9f403bb651f3949b5f63b40683917\
```

Comment:
this bootnode address should get from new build bootnode

Reply:
Or is this just example?

Reply:
no, it's just some random address. enode id doesn't matter for bootnode connection so I just used any available to simplify the process.